pkg/distributor/distributor.go

```diff
@@ -64,6 +64,10 @@ const (
 	typeMetadata = "metadata"
 
 	instanceIngestionRateTickInterval = time.Second
+
+	// mergeSlicesParallelism is a constant of how much go routines we should use to merge slices, and
+	// it was based on empirical observation: See BenchmarkMergeSlicesParallel
+	mergeSlicesParallelism = 8
```

Comment:
I think I am fine with it. Just one question, what if we make batch size a constant and get parallelism based on the input size? Then for small input we can still use 1 core and we use more cores for larger batch size.
We can still cap the concurrency at 8

Reply:
This is kinda done inside of the function:

`p := min(parallelism, len(a)/2)`

So, we will only use parallelism if the input is > 4 (and we will use only 2 cores in this case)

make sense?

Reply:
I see. Thanks for the explanation. Input length is basically the same as number of ingesters for the user. Then if the user has at least 16 ingesters, then we will use 8 parallelism for it?

16 ingesters sounds a small number to me. I am worried about having very small batch sizes. What about using a larger x below like 16 or even 32?  Is it better than 2 or you think it doesn't make much difference

```
p := min(parallelism, len(a)/x)
```

Reply:
Description updated with 16 ingesters and test case added.

WDYT?